src/v/kafka/data/cloud_topic_partition.cc

```diff
@@ -0,0 +1,599 @@
+    // Ask partition for its start offset
+    // TODO: ask dl_stm for its start offset
+    // the 'partition::sync_kafka_start_offset_override' is not invoked here
+    // because it's tied to both archival_metadata_stm and log_eviction_stm.
+    // For cloud topics we will do log eviction differently and the
+    // DeleteRecords API is not implemented yet. So the code is just
+    // using the start_offset of the Raft log at the moment which is incorrect.
+    auto so = _partition->raft_start_offset();
+    auto kso = _partition->get_offset_translator_state()->from_log_offset(so);
+    return kso;
+}
+
+ss::future<result<model::offset, error_code>>
+cloud_topic_partition::sync_effective_start(model::timeout_clock::duration) {
+    // TODO: ask dl_stm
+    co_return start_offset();
+}
+
+model::offset cloud_topic_partition::high_watermark() const {
+    auto ot_state = _partition->get_offset_translator_state();
+    return ot_state->from_log_offset(_partition->high_watermark());
+}
+
+checked<model::offset, error_code>
+cloud_topic_partition::last_stable_offset() const {
+    auto maybe_lso = _partition->last_stable_offset();
+    if (maybe_lso == model::invalid_lso) {
+        return error_code::offset_not_available;
+    }
+    auto ot_state = _partition->get_offset_translator_state();
+    return ot_state->from_log_offset(maybe_lso);
+}
+
+bool cloud_topic_partition::is_leader() const {
+    return _partition->is_leader();
+}
+
+ss::future<std::error_code> cloud_topic_partition::linearizable_barrier() {
+    auto r = co_await _partition->linearizable_barrier();
+    if (r) {
+        co_return raft::errc::success;
+    }
+    co_return r.error();
+}
+
+cluster::partition_probe& cloud_topic_partition::probe() {
+    return _partition->probe();
+}
+
+kafka::leader_epoch cloud_topic_partition::leader_epoch() const {
+    return leader_epoch_from_term(_partition->raft()->confirmed_term());
+}
+
+ss::future<storage::translating_reader> cloud_topic_partition::make_reader(
+  storage::log_reader_config cfg,
+  std::optional<model::timeout_clock::time_point> timeout) {
+    vassert(_ct_api != nullptr, "cloud topics api not initialized");
+
+    auto ot_state = _partition->get_offset_translator_state();
+
+    cfg.start_offset = ot_state->to_log_offset(cfg.start_offset);
+    cfg.max_offset = ot_state->to_log_offset(cfg.max_offset);
+    cfg.translate_offsets = storage::translate_offsets::yes;
+    cfg.type_filter = {model::record_batch_type::dl_placeholder};
+
+    // TODO: add code path that fetches metadata from the dl_stm for L1 read
+    // path
+
+    // This part comprises the control plane query. The partition is a metadata
+    // storage for the cloud topic.
+    auto underlying = co_await _partition->make_reader(cfg, timeout);
+    auto placeholders = co_await model::consume_reader_to_chunked_vector(
+      std::move(underlying), model::no_timeout);
+
+    // TODO: convert to a extent_meta
```

Comment:
What does this TODO mean? Can you elaborate?

Reply:
> What does this TODO mean? Can you elaborate?

@Lazin

Reply:
The TODO means that when we will have metadata store  we should update this code to fetch either from the partition or from the metadata store. For now it only fetches metadata from the underlying partition.